src/QirRuntime/public/IQuantumApi.hpp

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <complex>
+
+#include "CoreTypes.hpp"
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    struct IQuantumApi
+    {
+        virtual ~IQuantumApi() {}
+
+        // TODO: Only some simulators 'know' the state -- is it OK to have this method on IQuantumApi?
+        // The callback will be invoked on each basis vector (in the standard computational basis) in little-endian
+        // order until it returns `false` or the state is fully dumped.
+        typedef bool (*TGetStateCallback)(size_t /*basis vector*/, double /* amplitude Re*/, double /* amplitude Im*/);
+        virtual void GetState(TGetStateCallback callback) = 0;
+
+        virtual std::string DumpQubit(Qubit qubit) = 0;
+
+        // Qubit management
+        virtual Qubit AllocateQubit() = 0;
+        virtual void ReleaseQubit(Qubit qubit) = 0;
+
+        // Shortcuts
+        virtual void CNOT(Qubit control, Qubit target)

```

Comment:
Why is this alias needed?

Reply:
I don't feel strongly about those shortcuts and it's more here to collect feedback on various ways to shape the interface... I'd suggest to keep it for now so we can move on with the PR and file an issue that tracks finalizing the interface.